ablySpec/RealtimeClientChannel.swift

```diff
@@ -607,6 +607,55 @@ class RealtimeClientChannel: QuickSpec {
                     expect(Test.counter).toEventually(equal(2), timeout: testTimeout)
                 }
 
+                context("message cannot be decoded or decrypted") {
+
+                    // RTL7e
+                    it("should deliver with encoding attribute set indicating the residual encoding and error should be emitted") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+
+                        let channelOptions = ARTChannelOptions(encrypted: ARTCrypto.defaultParams())
+                        let channel = client.channels.get("test", options: channelOptions)
+
+                        let expectedMessage = ["key":1]
+                        let expectedData = try! NSJSONSerialization.dataWithJSONObject(expectedMessage, options: NSJSONWritingOptions(rawValue: 0))
+
+                        let transport = client.transport as! TestProxyTransport
+
+                        transport.beforeProcessingReceivedMessage = { protocolMessage in
+                            if protocolMessage.action == .Message {
+                                let messageReceived = protocolMessage.messages![0]
+                                // Replacement: `json/cipher+aes-128-cbc/base64` to `invalid/cipher+aes-128-cbc/base64`
+                                let newEncoding = "invalid" + messageReceived.encoding!.substringFromIndex("json".endIndex)
+                                messageReceived.encoding = newEncoding
+                            }
+                        }
+
+                        waitUntil(timeout: testTimeout) { done in
+                            let logTime = NSDate()
+
+                            channel.subscribe { message in
+                                // Last decoding failed: NSData -> JSON object, so...
+                                expect(message.data as? NSData).to(equal(expectedData))
+                                expect(message.encoding).to(equal("invalid"))
+
+                                let logs = querySyslog(forLogsAfter: logTime)
+                                let line = logs.reduce("") { $0 + "; " + $1 } //Reduce in one line
+                                expect(line).to(contain("ERROR: ARTDataDecoded failed to decode data as 'invalid'"))
```

Comment:
@tcard Rebased and now it's failing with an uncaught exception: `Unexpected exception raised: ARTRealtimeChannel: error decoding data: Error Domain=io.ably.cocoa Code=0 "decoding failed" UserInfo={NSLocalizedFailureReason=ARTErrorInfo with code 0, message: unknown encoding: 'bad_encoding_type', NSLocalizedDescription=decoding failed}`

https://github.com/ably/ably-ios/blob/66b38d811a09846c46c2c2cdffea3f7e3bb4e449/ably-ios/ARTRealtimeChannel.m#L329

But the spec says "_If there is an error decoding a message, the message is still delivered_" [RTL7d](http://docs.ably.io/client-lib-development-guide/features/#RTL7d)